src/protocol/boolean/bit_decomposition.rs

```diff
@@ -53,32 +53,31 @@ impl BitDecomposition {
 
         // Step 5. Add back [b] bitwise. [d]_B = BitwiseSum(c, [b]_B) where d ∈ Z
         //
-        // `BitwiseSum` outputs `l + 1` bits, so [d]_B is (l + 1)-bit long.
-        let d_b = bitwise_add_constant(ctx.narrow(&Step::AddBtoC), record_id, &r.b_b, c.as_u128())
-            .await?;
+        // `BitwiseSum` outputs one more bit than its input, so [d]_B is (el + 1)-bit long.
+        let d_b = add_constant(ctx.narrow(&Step::AddBtoC), record_id, &r.b_b, c.as_u128()).await?;
 
-        // Step 6. p <=? d. The paper says "p <? d", but should actually be "p <=? d"
+        // Step 6. p <? d.
         let q_p = BitwiseLessThanPrime::greater_than_or_equal_to_prime(
             ctx.narrow(&Step::IsPLessThanD),
             record_id,
             &d_b,
         )
         .await?;
 
-        // Step 7. a bitwise scalar value `f_B = bits(2^l - p)`
-        let l = u128::BITS - F::PRIME.into().leading_zeros();
-        let x = 2_u128.pow(l) - F::PRIME.into();
+        // Step 7. a bitwise scalar value `f_B = bits(2^el - p)`
+        let el = u128::BITS - F::PRIME.into().leading_zeros();
+        let x = (1 << el) - F::PRIME.into();
 
         // Step 8, 9. [g_i] = [q] * f_i
-        // Step 10. [h]_B = [d + g]_B, where [h]_B = ([h]_0,...[h]_(l+1))
+        // Step 10. [h]_B = [d + g]_B, where [h]_B = ([h]_0,...[h]_(el+1))
         //
-        // Again, `BitwiseSum` outputs `l + 1` bits. Since [d]_B is already
-        // `l + 1` bit long, [h]_B will be `l + 2`-bit long.
-        let h_b = bitwise_add_constant_maybe(ctx.narrow(&Step::AddDtoG), record_id, &d_b, x, &q_p)
-            .await?;
+        // Again, `BitwiseSum` outputs an extra bit. Since [d]_B is already
+        // `el + 1` bits long, [h]_B will be `el + 2`-bits long.
```

Comment:
...well it isn't any more. Perhaps update the comment to say something about how normal addition would result it i having two extra bits, so to avoid needless waste we are calling the "mod 2 el" version of this function.